docs/developer-guide/ksqldb-reference/aggregate-functions.md

```diff
@@ -281,11 +281,21 @@ Rows that have `col1` set to `NULL` are ignored.
 ## **`TOPK`**
 
 ```sql title="Applies to: stream<br>"
-TOPK(col1, k)
+TOPK(col1, otherCols..., k)
 ```
 
 Returns the Top *K* values for the given column and window.
 
+If only `col1` is provided, an array of values for that column is 
+
+returned. If `otherCols` are provided, a `STRUCT` is returned. The 
+
+`STRUCT` has a field named `sort_col` that contains value of `col1` in the 
+associated record. `otherCols` are in fields named `col0`, `col1`, `col2`, 

```

Comment:
im wondering now, given this constraint, does it make more sense to just return an array instead? @colinhicks @jnh5y

Reply:
If `otherCols` were instead an array, what would its type be?

Reply:
...i need to stop reviewing code at night

Reply:
We'd have to return a `List<Object>` to avoid using a struct, but ksql does not support returning `Object`s. Even if this was supported, I assume users would have to cast each value in the list, which still seems wonky to me. (Supporting `Object` as an aggregate type or return type is more complicated than allowing `Object` as a column argument.)